tree/ntupleutil/v7/src/RNTupleImporter.cxx

```diff
@@ -371,13 +372,29 @@ ROOT::RResult<void> ROOT::Experimental::RNTupleImporter::PrepareSchema()
 
 void ROOT::Experimental::RNTupleImporter::Import()
 {
-   if (fDestFile->FindKey(fNTupleName.c_str()) != nullptr)
+   TDirectory *targetDir = fDestFile.get();
+   auto parts = ROOT::Split(fNTupleName, "/", true);
+   if (parts.empty()) {
+      throw RException(R__FAIL("treeName cannot be empty"));
+   }
+   std::string treeName = parts.back();
```

Comment:
Rather than splitting and the rejoining the string you should be able to simply get the last index of `'/'` (using `strrchr`) and operate on the substrings you get from it, avoiding string copies.